src/mutex.rs

```diff
@@ -38,6 +38,23 @@ pub trait BusMutex<T> {
     fn lock<R, F: FnOnce(&T) -> R>(&self, f: F) -> R;
 }
 
+/// A dummy mutex for bus-sharing in a single task.
+///
+/// This mutex type can be used when all bus users are contained in a single execution context.  In
+/// such a situation, no actual mutex is needed, because a RefCell alone is sufficient to ensuring
+/// only a single peripheral can access the bus at the same time.
+pub struct NullMutex<T>(T);
```

Comment:
I know that the internal RefCell should make this `!Send` and `!Sync`, but it may be worth adding an explicit indication of this as well:
```
impl<T> !Send for NullMutext<T> {}
impl<T> !Sync for NullMutex<T> {}
```

Reply:
I'm afraid this is impossible on stable rust and we have to rely on the `!Send` from `RefCell` to propagate.  But I'll add a sentence to the docs to point it out explicitly.

Reply:
Sounds good to me! I was wondering if it wouldn't be possible, as the docs were a bit cryptic about it.